### PR TITLE
Remove old 'back' callback, and add dartdocs.

### PR DESCRIPTION
**Pull request by Hixie**
This adds some dartdocs to dart:ui, specifically for Window and friends.

### CODE REVIEW

File: sky/engine/core/dart/window.dart [hunk 1]
```diff
@@ -18,41 +24,102 @@ class WindowPadding {
...
-  Locale(this.languageCode, this.countryCode);
+  const Locale(this.languageCode, this.countryCode);
 
+  /// The primary language subtag for the locale.
   final String languageCode;
+
+  /// The region subtag for the locale.
   final String countryCode;
 
+  bool operator ==(dynamic other) {
```
Comment (eseidelGoogle) [inline sky/engine/core/dart/window.dart]: It's so dumb/error-prone we hand-roll these.  I know we considered quiver and decided against it for now.   Le sigh.

File: sky/engine/core/dart/window.dart [hunk 2]
```diff
@@ -18,41 +24,102 @@ class WindowPadding {
...
+  /// callback be invoked.
   void scheduleFrame() native "Window_scheduleFrame";
+
+  /// Updates the application's rendering on the GPU with the newly provided
+  /// [Scene]. For optimal performance, this should only be called in response
+  /// to the [onBeginFrame] callback being invoked.
   void render(Scene scene) native "Window_render";
 }
 
+/// The [Window] singleton.
```
Comment (eseidelGoogle) [inline sky/engine/core/dart/window.dart]: I love all the docs.

**Comment by eseidelGoogle** [comment]
lgtm, but travis is red :(

File: sky/engine/core/dart/window.dart (hunk 1 above)
Comment (Hixie) [inline sky/engine/core/dart/window.dart]: Quiver was never going to help with the `operator ==`s. I don't think there's anything we can really do about those. At least we've established a pattern, though.

For the hashCodes, I am working on a better solution, see #88.

**Comment by Hixie** [comment]
Oops, forgot to remove onEvent's typedef, but travis caught it!

File: sky/engine/core/dart/window.dart (hunk 1 above)
Comment (eseidelGoogle) [inline sky/engine/core/dart/window.dart]: It's just silly that we need to write out this boilerplate every time for equals.  w/o mirrors we can't easily inspect what members this class has, I presume.  Add a field and forget to add it to equals or hashcode and you're a sad panda.

Comment (Hixie) [inline sky/engine/core/dart/window.dart]: Yeah. I have no good ideas for simplifying `operator==`.

Comment (abarth) [inline sky/engine/core/dart/window.dart]: Language support could help us.  The VM could provide an extensional equals function that's a dual to `identical`.

File: sky/engine/core/dart/window.dart [hunk 3]
```diff
@@ -18,41 +23,102 @@ class WindowPadding {
...
   double _devicePixelRatio;
 
+  /// The dimensions of the rectangle into which the application will be drawn,
+  /// in logical pixels.
   Size get size => _size;
   Size _size;
 
+  /// The number of pixels on each side of the display rectangle into which the
+  /// application can render, but over which the operating system will likely
+  /// place system UI (such as the Android system notification area).
```
Comment (abarth) [inline sky/engine/core/dart/window.dart]: Another good example is TVs which let you draw into an "overdraw" region that often isn't actually visible.

File: sky/engine/core/dart/window.dart [hunk 4]
```diff
@@ -18,41 +23,102 @@ class WindowPadding {
...
+  /// A callback that is invoked when pointer data is available. The data is
+  /// provided in the form of a raw byte stream containing an encoded mojo
+  /// PointerPacket.
   _PointerPacketCallback onPointerPacket;
-  VoidCallback onMetricsChanged;
-  VoidCallback onLocaleChanged;
 
+  /// The route or path that the operating system requested when the application
+  /// was launched.
   String defaultRouteName;
```
Comment (abarth) [inline sky/engine/core/dart/window.dart]: Do we still want this to be mutable?

Comment (Hixie) [inline sky/engine/core/dart/window.dart]: I don't know, I missed that it was. It has an odd name for a globally-mutable field.

**Pull request merged**